### PR TITLE
add support for node-selector

### PR DESCRIPTION
**Pull request by dschiemann80**
- introduce cmdline arg and docker label for setting a node selector
- format is a comma-separated list of key-value pairs

I tried to conform to kubedock conventions as best as I could. I wasn't sure where to put the parsing code, so I went with containers.go as a first try.
Please let me know if you find this useful and if I should adapt anything.

### CODE REVIEW

**Comment by joyrex2001** [comment]
Thanks appreciated, sorry for the late response.

I aggree with the suggested format, and that it should be configureable via a label as well as cli arguments. Note that the applying a node selector should also be possible with the use of a pod-template. However, in this case it can be dynamically applied (by using the labels) as well, which is a nice feature to have.

Generally, if such a label is introduced, the processing of the label is handled in `internal/model/types/container.go` and the routes implementations (`docker/containers.go` and `libpod/containers.go`) make sure the label is present (as you already did for the docker variant). For example, see how the pull policy is implemented, where the label is populated, and the value is processed and retrieved in the model implementation with GetImagePullPolicy. The node-selector should be implemented in a similar fashion.

Can you move the code to the model, and add some unit tests as well?
Can you also update the documentation in the `README.md` and `config.md` files?

**Comment by dschiemann80** [comment]
Thanks for your feedback.

I moved the code to the model as you suggested and added the label handling to the libpod routes. I am now honoring the node selector from the pod template by passing it into GetNodeSelector() as an argument.
I also added unit tests and documentation.

Please let me know if you would like to have adapted anything else.

File: internal/model/types/container.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type Container struct {
 	MappedPorts    map[int]int
 	Networks       map[string]interface{}
 	NetworkAliases []string
+	NodeSelector   map[string]string
```
Comment (joyrex2001) [inline internal/model/types/container.go:40]: This isn't used anymore.

File: internal/server/routes/docker/containers.go [hunk 2]
```diff
@@ -91,6 +94,7 @@ func ContainerCreate(cr *common.ContextRouter, c *gin.Context) {
 		Binds:        in.HostConfig.Binds,
 		Mounts:       mounts,
 		PreArchives:  []types.PreArchive{},
+		NodeSelector: map[string]string{},
```
Comment (joyrex2001) [inline internal/server/routes/docker/containers.go:97]: Same, this isn't used anymore.

File: internal/server/routes/libpod/containers.go [hunk 3]
```diff
@@ -74,6 +77,7 @@ func ContainerCreate(cr *common.ContextRouter, c *gin.Context) {
 		ExposedPorts: map[string]interface{}{},
 		ImagePorts:   map[string]interface{}{},
 		Labels:       in.Labels,
+		NodeSelector: map[string]string{},
```
Comment (joyrex2001) [inline internal/server/routes/libpod/containers.go:80]: And this one as well.

**Review by joyrex2001 — CHANGES_REQUESTED** [review]
Can you remove the unused code?

**Comment by dschiemann80** [comment]
Sure, removed it.

**Comment by joyrex2001** [comment]
Awesome, thanks @dschiemann80 !

**Pull request merged**